src/util/strencodings.h

```diff
@@ -67,6 +68,66 @@ std::vector<Byte> ParseHex(std::string_view hex_str)
 {
     return TryParseHex<Byte>(hex_str).value_or(std::vector<Byte>{});
 }
+
+/** consteval version of HexDigit() without the lookup table. */
+consteval uint8_t ConstevalHexDigit(const char c)
+{
+    if (c >= '0' && c <= '9') return c - '0';
+    if (c >= 'a' && c <= 'f') return c - 'a' + 0xA;
+
+    throw "Only lowercase hex digits are allowed, for consistency";
+}
+
+struct ConstevalHexLiteral {
+    const std::string_view inner;
+    template <size_t N>
+    consteval ConstevalHexLiteral(const char (&hex_str)[N]) : inner{hex_str} { Validate(inner); }
+    consteval ConstevalHexLiteral(const std::string_view hex_str) : inner{hex_str} { Validate(inner); }
+    consteval ConstevalHexLiteral(std::nullptr_t) = delete;
+
+    static consteval void Validate(const std::string_view hex_str)
+    {
+        std::ranges::for_each(hex_str, ConstevalHexDigit);
+
+        if (hex_str.size() % 2 == 1) throw "2 hex digits required per byte";
+    }
+};
+
+/** Validates input at compile time, but allocates the vector at runtime in
+ *  order to cross the compile time/runtime dynamic memory barrier.
+ */
+template <typename Byte = uint8_t>
+std::vector<Byte> VecFromHex(ConstevalHexLiteral hex_str)
```

Comment:
Nit: "vec" vs "array" (abbreviation vs full) -> can we either name them "VecFromHex" and "ArrFromHex" or rather "VectorFromHex" and "ArrayFromHex" (or even shorter: "HexToVector" and "HexToArray"?